packages/generator/src/service-base-path.ts

```diff
@@ -0,0 +1,52 @@
+import { parse } from 'path';
+import { createLogger } from '@sap-cloud-sdk/util';
+import { ServiceOptions } from '@sap-cloud-sdk/generator-common/internal';
+import { basePathFromSwagger } from './swagger-parser/swagger-util';
+import { ServiceMetadata } from './edmx-parser/edmx-file-reader';
+
+const logger = createLogger({
+  package: 'generator',
+  messageContext: 'options-per-service'
+});
+
+/**
+ * @internal
+ */
+export function getBasePath(
+  metadata: ServiceMetadata,
+  skipValidation: boolean,
+  optionsPerServiceIn: ServiceOptions
+): string {
+  let basePath =
+    optionsPerServiceIn?.basePath ||
+    basePathFromSelfLink(metadata.edmx.selfLink) ||
+    basePathFromSwagger(metadata.swagger);
+  if (!basePath) {
+    if (skipValidation) {
+      logger.warn(
+        `[ ${
+          parse(metadata.edmx.path.toString()).name
+        } ] No base path could be determined from available metadata! Setting "basePath" to "/" in the "options-per-service.json".`
+      );
+      basePath = '/';
+    } else {
+      throw new Error(
+        `[ ${
+          parse(metadata.edmx.path.toString()).name
```

Comment:
I think this error message could be more user friendly. How is this supposed to work? I just tried to put myself into the shoes of a user, downloaded an edmx file, started the generator, got this message.

How should I proceed now? What I did was to set the skipValidation flag to let me generate a options-per-service.json file. Would not this be a good default behaviour so that the user does not manually need to create a options-per-service.json file? What are we expecting the user to do?

Reply:
Changed the order so that the options are written even if the generation fails. Tried to improve the error message as well - I would like to avoid scope creep on the PR :-).